users/views.py

```diff
@@ -1,3 +1,59 @@
-from django.shortcuts import render
+import json, re, bcrypt
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+
+from .models import User
+
+
+class SignUpView(View):
+    def post(self, request):
+        try:
+            data           = json.loads(request.body)
+            name           = data['name']
+            email          = data['email']
+            password       = data['password']
+            mobile_number  = data['mobile_number']
+            birthday       = data.get('birthday')
+
+            REGEX_PASSWORD = '^(?=.*[A-Za-z])(?=.*\d)(?=.*[?!@#$%*&])[A-Za-z\d?!@#$%*&]{8,}$'
+            REGEX_EMAIL    = '^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
+            REGEX_BIRTHDAY = '^(19[0-9][0-9]|20\d{2})-(0[0-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])$'
+            REGEX_MOBILE   = '^\d{3}-\d{3,4}-\d{4}$'
+
+            checkRegex(REGEX_EMAIL, email)
+
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({'MESSAGE': 'ALREADY_EXISTS_EMAIL'}, status = 400)
+            
+            checkRegex(REGEX_PASSWORD, password)
+                
+            if User.objects.filter(mobile_number=mobile_number).exists():
+                return JsonResponse({'MESSAGE': 'ALREADY_EXISTS_MOBILE_NUMBER'}, status = 400)
+
+            checkRegex(REGEX_MOBILE, mobile_number)
+
+            if birthday: 
+                if not re.compile(REGEX_BIRTHDAY).match(birthday):
+                    raise ValueError
```

Comment:
python은 interpreter 언어이기 때문에 코드를 한줄씩 읽으며 실행합니다. 

그렇기에 input으로 들어오는 값들에 대한 검증이 먼저 선행된 뒤 통과가 됐을 때,
비로소 분기된 로직(데이터베이스를 조회하는)을 검증하도록 작성하는것이 더 낫습니다.

```suggestion
            checkRegex(REGEX_EMAIL, email)
            checkRegex(REGEX_PASSWORD, password)
            checkRegex(REGEX_MOBILE, mobile_number)
            
            if birthday: 
                checkRegex(REGEX_BIRTHDAY, birthday)

            if User.objects.filter(email=email).exists():
                return JsonResponse({'MESSAGE': 'ALREADY_EXISTS_EMAIL'}, status = 400)
            
            if User.objects.filter(mobile_number=mobile_number).exists():
                return JsonResponse({'MESSAGE': 'ALREADY_EXISTS_MOBILE_NUMBER'}, status = 400)
```

Reply:
우오..28,29 부분은 놀라워요
vaild에러를 반환하는 함수를 어떻게 조건문에 이용할까 고민하다가 그냥 썼는데 파이썬을 다시 배워야할것같아요.....